### ISSUE TITLE
False positive: [UPM_UNCALLED_PRIVATE_METHOD] on a default method calling a private method?

### ISSUE THREAD

**Issue by Burger0419** [spotbugs/spotbugs#3801]
I define a private method named 'retrySearchFiles( )' and a default method named 'searchFiles( )'. The 'searchFiles( )' calls the 'retrySearchFiles( )' as follows: 

default List<String> searchFiles() {
    ... ...
    List<String> result = retrySearchFiles();
    ... ...
}

private List<String> retrySearchFiles() {
    ... ...
}

Then I use spotbugs to scan it and get a defect says 'Private method com.xxx.IFtpDownload.retrySearchFiles() is never called'

**Comment by JuditKnoll** [comment]
I tried to reproduce this issue locally with this code example:
```java
import java.util.ArrayList;
import java.util.List;

public interface Issue3801 {
    default List<String> foo() {
        List<String> result = bar();
        return result;
    }

    private List<String> bar() {
        return new ArrayList<String>();
    }
}
```

There was no `UPM_UNCALLED_PRIVATE_METHOD` bug reported for me.
Can you please share a reproducible minimal, complete example?
What kind of compiler are you using?

**Comment by gtoison** [comment]
This might be a duplicate of https://github.com/spotbugs/spotbugs/issues/1988 if you're using an older version of SpotBugs
